config/crd/bases/cpi.tanzu.vmware.com_vspherecpiconfigs.yaml

```diff
@@ -106,6 +106,7 @@ spec:
                         - kind
                         - name
                         type: object
+                        x-kubernetes-map-type: atomic
```

Comment:
I think this is less to do with concurrent modifications and more to do with "merging" fields [using server-side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy). `atomic` means you are replacing the whole map instead of specific fields.

Reply:
Concurrency control is always enforced using resourceVersion comparison.

Reply:
I see, thanks for the info. Looks like `atomic` will restrict other managers from updating fields. If the current default behavior is `granular` I think we should keep that.

Reply:
Default is `granular`, but these fields are specifically inheriting `atomic` from `corev1` structs they refer to.

https://github.com/vmware-tanzu/tanzu-framework/blob/5c69b9e7aba658ced42b6df3298d6a4b4e713ab3/apis/cpi/v1alpha1/vspherecpiconfig_types.go#L115

https://github.com/kubernetes/api/blob/f5e1938afa507ee250e4a0a7a45cd2498e3750d3/core/v1/types.go#L5690-L5714